Tasks/AzureRmWebAppDeployment/azurermwebappdeployment.ts

```diff
@@ -45,6 +49,46 @@ async function run() {
         var isFolderBasedDeployment = await isInputPkgIsFolder(webDeployPkg);
         var publishingProfile = await azureRmUtil.getAzureRMWebAppPublishProfile(SPN, webAppName, resourceGroupName, deployToSlotFlag, slotName);
         tl._writeLine(tl.loc('GotconnectiondetailsforazureRMWebApp0', webAppName));
+        
+        if(xmlTransformationAndVariableSubstitution && xmlTransformation) {
+            if(canUseWebDeploy(useWebDeploy)) {
+                var tempPackagePath;
+                if(!isFolderBasedDeployment) {
+                    tempPackagePath = path.join(tl.getVariable('System.DefaultWorkingDirectory'), '_temp_package_path');
+                    if(tl.exist(tempPackagePath)){
+                        tl.rmRF(tempPackagePath);
+                    }
+                    tl._writeLine("Unzipping the package at location : " + tempPackagePath);
+                    compressor.unzip(webDeployPkg, tempPackagePath);
+                }
+                else {
+                    tempPackagePath = tl.getVariable('System.DefaultWorkingDirectory');
+                    tl._writeLine("Copying the package at location : " + tempPackagePath);
+                    tl.cp(webDeployPkg, tempPackagePath, "-rf");
+                    tempPackagePath = path.join(tempPackagePath, path.win32.basename(webDeployPkg));
```

Comment:
Test with folder and zip both and ensure after transformation webapp is working fine.